### PR TITLE
generate id from given timestamp

### PR DESCRIPTION
**Pull request by hanskoff**

### CODE REVIEW

File: idGenerator/src/main/scala/com/softwaremill/id/IdGenerator.scala [hunk 1]
```diff
@@ -22,6 +23,12 @@ class DefaultIdGenerator(workerId: Long = 1, datacenterId: Long = 1) extends IdG
       idWorker.nextId()
     }
   }
+
+  def idFrom(timestamp: Long): Long = {
+    synchronized {
```
Comment (adamw) [inline idGenerator/src/main/scala/com/softwaremill/id/IdGenerator.scala]: does it have to be synchronized?

File: idGenerator/src/main/scala/com/softwaremill/id/IdGenerator.scala [hunk 2]
```diff
@@ -7,6 +7,7 @@ import com.typesafe.scalalogging.StrictLogging
   */
 trait IdGenerator {
   def nextId(): Long
+  def idFrom(timestamp: Long): Long
```
Comment (adamw) [inline idGenerator/src/main/scala/com/softwaremill/id/IdGenerator.scala]: it's not really an id (one that could be ever generated), maybe `idBaseAt`?

File: idGenerator/src/test/scala/com/softwaremill/id/DefaultIdGeneratorSpec.scala [hunk 3]
```diff
@@ -0,0 +1,147 @@
+package com.softwaremill.id
+import org.scalatest.{MustMatchers, WordSpec}
+
+class DefaultIdGeneratorSpec extends WordSpec with MustMatchers {
```
Comment (adamw) [inline idGenerator/src/test/scala/com/softwaremill/id/DefaultIdGeneratorSpec.scala]: thanks for the tests :)

File: idGenerator/src/test/scala/com/softwaremill/id/DefaultIdGeneratorTest.scala [hunk 4]
```diff
@@ -0,0 +1,53 @@
+package com.softwaremill.id
+import org.scalatest.{FunSuite, Matchers}
+
+class DefaultIdGeneratorTest extends FunSuite with Matchers {
```
Comment (adamw) [inline idGenerator/src/test/scala/com/softwaremill/id/DefaultIdGeneratorTest.scala]: why are the tests split into 2 files?

File: idGenerator/src/main/scala/com/softwaremill/id/IdGenerator.scala (hunk 1 above)
Comment (hanskoff) [inline idGenerator/src/main/scala/com/softwaremill/id/IdGenerator.scala]: I doesn't have to be, fixing

File: idGenerator/src/main/scala/com/softwaremill/id/IdGenerator.scala (hunk 2 above)
Comment (hanskoff) [inline idGenerator/src/main/scala/com/softwaremill/id/IdGenerator.scala]: let it be `idBaseAt`

File: idGenerator/src/test/scala/com/softwaremill/id/DefaultIdGeneratorTest.scala (hunk 4 above)
Comment (hanskoff) [inline idGenerator/src/test/scala/com/softwaremill/id/DefaultIdGeneratorTest.scala]: merging into one

**Comment by adamw** [comment]
Released 1.1.0

**Comment by hanskoff** [comment]
thnx

**Pull request merged**